### PR TITLE
Turn on code-cov for PR checking of unit test coverage

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This enables code cov PR checks and commenting. The idea of this configuration is not to have a specific overall target but to just have each PR increase unit test coverage. Code cov automatically determines what the increase should be.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

There isn't a way to test every aspect of this config, but you can see that the valid config produces PR comments and CI output that passes. We can evaluate over time if this configuration meets our needs.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
-  status:
-    project:
-      default:
-        informational: true
-    patch:
-      default:
-        informational: true
-    changes: off
+coverage:
+    precision: 2
```
Comment (lsinger) [inline .codecov.yml]: I think `1` or even `0` might be sufficiently precise here.

File: .codecov.yml [hunk 2]
```diff
@@ -1,27 +1,46 @@
...
+    round: nearest
+    range: '50...100'
```
Comment (lsinger) [inline .codecov.yml]: Looking at their docs, the first number defines at what point the graph is "red" and the second number determines at what coverage we reach "green". IMHO `80` is already an amazing coverage that might deserve being shown in green. ☺️

**Review by lsinger — DISMISSED** [review]
Left two notes but I think as is this should already be a good foundation to iterate on as needed. 

I'll let you do the merge because it's the end of my day and I think we should be present to monitor things for a while.

**Comment by samueljseay** [comment]
@lsinger sadly it was a red herring lol, the codecov infrastructure was down for maintenance the first time the checks tried to run.

**Comment by samueljseay** [comment]
@lsinger the settings are looking good now and working as expected. I added your suggestions. OK for a re-review?

**Review by lsinger — APPROVED** [review]

**Comment by lsinger** [comment]
LGTM! 🚢

**Pull request merged**